src/chainstate/stacks/boot/bns.clar

```diff
@@ -0,0 +1,973 @@
+      ;; The name's namespace must be launched
+      (asserts!
+        (is-some (get launched-at namespace-props))
+        (err ERR_NAMESPACE_NOT_LAUNCHED))
+      ;; The preorder must have been created after the launch of the namespace
+      (asserts!
+        (> (get created-at preorder) (unwrap! (get launched-at namespace-props) (err ERR_PANIC)))
+        (err ERR_NAME_PREORDERED_BEFORE_NAMESPACE_LAUNCH))
+      ;; The preorder entry must be unclaimed - todo(ludo): is this assertion redundant?
+      (asserts!
+        (is-eq (get claimed preorder) false)
+        (err ERR_NAME_ALREADY_CLAIMED))
+      ;; Less than 24 hours must have passed since the name was preordered
+      (asserts!
+        (< block-height (+ (get created-at preorder) NAME_PREORDER_CLAIMABILITY_TTL))
+        (err ERR_NAME_CLAIMABILITY_EXPIRED))
+      ;; The amount burnt must be equal to or greater than the cost of the name
+      (asserts!
+        (>= (get stx-burned preorder) (compute-name-price name (get price-function namespace-props)))
+        (err ERR_NAME_STX_BURNT_INSUFFICIENT))
+      ;; The principal can register a name
+      (asserts!
+        can-sender-register-name
+        (err ERR_PRINCIPAL_ALREADY_ASSOCIATED))
+      ;; Mint the name if new, transfer the name otherwise.
+      (if (is-none current-owner)
+        (begin
+          (unwrap! 
+            (nft-mint? 
+              names 
+              (tuple (namespace namespace) (name name)) 
+              contract-caller)
+            (err ERR_NAME_COULD_NOT_BE_MINTED))
+          (map-set owner-name
+            ((owner contract-caller))
+            ((namespace namespace) (name name))))
+        (if (is-eq contract-caller (unwrap! current-owner (err ERR_PANIC)))
+          true
+          (let ((previous-owner (unwrap! current-owner (err ERR_PANIC)))) 
+            (unwrap!
+              (update-name-ownership? namespace name previous-owner contract-caller)
+              (err ERR_NAME_COULD_NOT_BE_TRANSFERED)))))
+      ;; Update name's metadata / properties
+      (update-zonefile-and-props
+        namespace 
+        name  
+        (some block-height)
+        none
+        none
+        zonefile-hash)
+      (ok true))))
+
+;; NAME_UPDATE
+;; A NAME_UPDATE transaction changes the name's zone file hash. You would send one of these transactions 
+;; if you wanted to change the name's zone file contents. 
+;; For example, you would do this if you want to deploy your own Gaia hub and want other people to read from it.
+(define-public (name-update (namespace (buff 19))
+                            (name (buff 16))
+                            (zonefile-hash (buff 20)))
+  (let (
+    (owner (unwrap!
+      (nft-get-owner? names (tuple (name name) (namespace namespace)))
+      (err ERR_NAME_NOT_FOUND))) ;; The name must exist
+    (name-props (unwrap!
+      (map-get? name-properties ((name name) (namespace namespace)))
+      (err ERR_NAME_NOT_FOUND)))) ;; The name must exist
```

Comment:
You need to also check that the namespace has been launched.  `name-update`s shouldn't be allowed on names whose namespaces are sill being launched.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/2013/commits/5241bdf61b21d7f855df709cfd157a89a7147a0b